backend/src/__test__/controllers/questionController.test.ts

```diff
@@ -0,0 +1,117 @@
+import request from 'supertest';
+import mongoose from 'mongoose';
+import { MongoMemoryServer } from 'mongodb-memory-server';
+import app from '../../server'; // Import the Express app
+import Question, { IQuestion } from '../../models/questionModel';
+
+// Setup MongoDB memory server before tests
+let mongoServer: any;
+
+beforeAll(async () => {
+  mongoServer = await MongoMemoryServer.create();
+  const uri = mongoServer.getUri();
+  await mongoose.connect(uri);
+});
+
+afterAll(async () => {
+  await mongoose.connection.dropDatabase();
+  await mongoose.connection.close();
+  await mongoServer.stop();
+});
+
+// Clear the database after each test
+afterEach(async () => {
+  await Question.deleteMany({});
+});
+
+describe('Question Controller', () => {
+  // Test: Create a new question
+  it('should create a new question', async () => {
+    const newQuestion: Partial<IQuestion> = {
+      questionId: 1,
+      title: 'What is Node.js?',
+      description: 'Node.js is a runtime for JavaScript.',
+      category: 'Programming',
+      complexity: 'Easy',
+    };
+
+    const response = await request(app)
+      .post('/api/questions')
+      .send(newQuestion);
+
+    expect(response.status).toBe(201);
+    expect(response.body.title).toBe(newQuestion.title);
+  });
+
+  // Test: Fetch all questions
+  it('should fetch all questions', async () => {
+    const questions = [
+      { questionId: 1, title: 'Q1', description: 'Desc1', category: 'Cat1', complexity: 'Easy' },
+      { questionId: 2, title: 'Q2', description: 'Desc2', category: 'Cat2', complexity: 'Medium' }
+    ];
+    await Question.insertMany(questions);
+
+    const response = await request(app).get('/api/questions');
+    expect(response.status).toBe(200);
+    expect(response.body.length).toBe(2);
+  });
+
+  // Test: Fetch a question by ID
+  it('should fetch a question by ID', async () => {
+    const question = new Question({
+      questionId: 1,
+      title: 'What is JavaScript?',
+      description: 'A programming language.',
+      category: 'Programming',
```

Comment:
should be categories, not category (example questions given in document have multiple categories)